web/src/beta/features/Assets/AssetHooks/hooks.tsx

```diff
@@ -0,0 +1,174 @@
+import { useCallback, useState, useRef, useMemo, useEffect } from "react";
+
+import { Asset, SortType } from "@reearth/beta/features/Assets/types";
+import { autoFillPage, onScrollToBottom } from "@reearth/beta/utils/infinite-scroll";
+import { useAssetsFetcher } from "@reearth/services/api";
+import { Maybe, AssetSortType as GQLSortType } from "@reearth/services/gql";
+import { useT } from "@reearth/services/i18n";
+
+const assetsPerPage = 20;
+
+const enumTypeMapper: Partial<Record<GQLSortType, string>> = {
+  [GQLSortType.Date]: "date",
+  [GQLSortType.Name]: "name",
+  [GQLSortType.Size]: "size",
+};
+
+function toGQLEnum(val?: SortType) {
+  if (!val) return;
+  return (Object.keys(enumTypeMapper) as GQLSortType[]).find(k => enumTypeMapper[k] === val);
+}
+
+function pagination(
+  sort?: { type?: Maybe<SortType>; reverse?: boolean },
+  endCursor?: string | null,
+) {
+  const reverseOrder = !sort?.type || sort?.type === "date" ? !sort?.reverse : !!sort?.reverse;
+
+  return {
+    after: !reverseOrder ? endCursor : undefined,
+    before: reverseOrder ? endCursor : undefined,
+    first: !reverseOrder ? assetsPerPage : undefined,
+    last: reverseOrder ? assetsPerPage : undefined,
+  };
+}
+
+export default ({
+  workspaceId,
+}: {
+  workspaceId?: string;
+  onAssetSelect?: (inputValue?: string) => void;
+}) => {
+  const [sort, setSort] = useState<{ type?: SortType; reverse?: boolean }>();
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const [selectedAssets, selectAsset] = useState<Asset[]>([]);
+  const isGettingMore = useRef(false);
+
+  const { useAssetsQuery, useRemoveAssets } = useAssetsFetcher();
+
+  const { assets, hasMoreAssets, loading, isRefetching, endCursor, fetchMore } = useAssetsQuery({
+    teamId: workspaceId ?? "",
+    pagination: pagination(sort),
+    sort: toGQLEnum(sort?.type),
+    keyword: searchTerm,
+  });
+
+  const t = useT();
+  const [deleteModalVisible, setDeleteModalVisible] = useState(false);
+  const [localSearchTerm, setLocalSearchTerm] = useState<string>(searchTerm ?? "");
+  const openDeleteModal = useCallback(() => setDeleteModalVisible(true), []);
+  const closeDeleteModal = useCallback(() => setDeleteModalVisible(false), []);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const sortOptions: { key: string; label: string }[] = useMemo(
+    () => [
+      { key: "date", label: t("Last Uploaded") },
+      { key: "date-reverse", label: t("First Uploaded") },
+      { key: "name", label: t("A To Z") },
+      { key: "name-reverse", label: t("Z To A") },
+      { key: "size", label: t("Size Small to Large") },
+      { key: "size-reverse", label: t("Size Large to Small") },
+    ],
+    [t],
+  );
+
+  const iconChoice =
```

Comment:
useMemo

Reply:
done , note I think we won't use it , I will check with red if it will be ever used

Reply:
Yeah I just checked the code and it seems not to be used at the moment. If red said it'll be used, comment it out, or else delete it please 🙏🏻

Reply:
I checked the design and it is not exist any more , so I removed it